Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -378,12 +378,23 @@ ErrorOr<void> VirtualFileSystem::mount_root(FileSystem& fs)
     return {};
 }
 
-auto VirtualFileSystem::find_mount_for_host(InodeIdentifier id) -> Mount*
+auto VirtualFileSystem::find_mount_for_host_custody(Custody const& current_custody) -> Mount*
 {
     return m_mounts.with([&](auto& mounts) -> Mount* {
-        for (auto& mount : mounts) {
-            if (mount.host() && mount.host()->identifier() == id)
-                return &mount;
+        // NOTE: We either search for the root mount or for a mount that has a parent custody!
+        if (!current_custody.parent()) {
+            for (auto& mount : mounts) {
+                if (!mount.host_custody())
+                    return &mount;
+            }
+            // NOTE: There must be a root mount entry, so fail if we don't find it.
+            VERIFY_NOT_REACHED();
```

Comment:
Is this a hard invariant that always hold true? I'm also fine with returning `nullptr` / `ENODEV` below.

Reply:
Yes, it always hold true. Other code depends on it, so I guess this is fine.